### PR TITLE
Fix FuzzMongoRead

### PR DESCRIPTION
**Pull request by smallinsky**

### CODE REVIEW

**Review by Tener — APPROVED** [review]
Any plans to upstream this fix?

**Review by greedy52 — APPROVED** [review]

**Comment by Tener** [comment]
@smallinsky Can we merge this?

**Comment by smallinsky** [comment]
@ping @zmb3

File: lib/srv/db/mongodb/protocol/util.go [hunk 1]
```diff
@@ -48,3 +48,19 @@ func appendCString(b []byte, str string) []byte {
...
+
+	return (int32(src[0]) | int32(src[1])<<8 | int32(src[2])<<16 | int32(src[3])<<24), src[4:], true
+}
+
+func readcstring(src []byte) (string, []byte, bool) {
+	idx := bytes.IndexByte(src, 0x00)
+	if idx < 0 {
+		return "", src, false
+	}
+	return string(src[:idx]), src[idx+1:], true
```
Comment (zmb3) [inline lib/srv/db/mongodb/protocol/util.go:65]:
```suggestion
	before, after, found := bytes.Cut(src, []byte{0x00})
	return string(before), after, found
```

This whole thing could probably be simplified with bytes.Cut.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**